### PR TITLE
Reload log_async only when it changed

### PR DESCRIPTION
**Pull request by icristescu**

### CODE REVIEW

File: src/index.ml [hunk 1]
```diff
@@ -363,18 +363,25 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
...
     ( match t.log with
     | None -> t.log <- try_load_log t (log_path t.root)
     | Some _ -> () );
-    may (fun log -> IO.close log.io) t.log_async;
-    t.log_async <- try_load_log t (log_async_path t.root);
+    ( match t.log_async with
+    | None -> t.log_async <- try_load_log t (log_async_path t.root)
+    | Some log ->
+        let log_offset = IO.offset log.io in
+        let new_log_offset = IO.force_offset log.io in
```
Comment (pascutto) [inline src/index.ml]: I think it's possible that `log_async` is not present anymore and this instruction will fail.

File: src/index.ml [hunk 2]
```diff
@@ -363,18 +363,25 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
...
+        if log_offset <> new_log_offset then (
+          Tbl.clear log.mem;
+          iter_io (add_log_entry log) log.io ) );
```
Comment (pascutto) [inline src/index.ml]: What if the new log offset is smaller than the previous one? That would mean that it has been reset in the meanwhile, but this hashtbl will be grow indefinitely.

File: src/index.ml (hunk 1 above)
Comment (icristescu) [reply src/index.ml]: Yes, indeed. I can wrap it in a catch exception

File: src/index.ml (hunk 2 above)
Comment (icristescu) [reply src/index.ml]: I'm not sure I understand, there is a tbl clear just before.

**Pull request merged**